r2/r2/lib/menus.py

```diff
@@ -609,7 +605,7 @@ def make_title(self, attr):
 class SearchSortMenu(SortMenu):
     """Sort menu for search pages."""
     _default = 'relevance'
-    mapping = search_sorts
+    mapping = g.search.sorts
```

Comment:
This line is causing problems; since class variables like this get created at module import time, it's trying to read from `g` at import time, which can be before pylons has configured `g`. Even if it works now or in your environment, all it takes is a change in import ordering, or a plugin importing things differently, to trigger unresolvable `AttributeError`s on app start up.

Looking at the other code: Nothing appears to rely on `mapping` or `_options` being available at the class level, so these should be defined as `property`s instead, so that `g.search.sorts` is read at run-time rather than at import time.